### DISCUSSION TITLE
User registration in the user application

### DISCUSSION THREAD

**Discussion by thiagow** [Budibase/budibase#7459] [General]
Can I have a user registration screen from within my app?

I would also like to know if I can have users self-registered in my application?

**Comment by R2bEEaton** [comment]
Hi @thiagow,

This won't be the most helpful answer, but... yes. Look into the API docs for creating users: https://docs.budibase.com/reference/post_users. The trick will be sending the request without revealing your API key to the public. Perhaps it could be done with a Bash script in Automations, but the bash shell is incredibly limited (no `curl` or `wget`).

Budibase has a focus on internal applications, so something like you're asking will probably never be added officially, and is therefore going to be difficult to do yourself.

**Comment by alcompstudio** [comment]
But this is already bad ... Public registration of users for created applications is very necessary. For example, I want to make a portal for Customers and Contractors (something like a Job Search), where users can register in the desired role and have access to work inside the portal.
I am not a programmer, I really hoped that there would be such an opportunity for no-code application. A very powerful platform, but such a common feature as public user registration is not available.

**Reply by shogunpurple** [reply]
We've answered this several times throughout our forums and discord:

Internal application are what budibase is built for and specialises in - it's not a technical/scaling limitation, just not what budibase is best at.

If you are able to build an external app and get it to work for you then go ahead, but there's likely to be a few features you would want with a SaaS that we won't build because they don't apply to internal apps.

More info here: https://github.com/Budibase/budibase/discussions/3866

**Reply by alcompstudio** [reply]
I understand, thanks for the warning. So this platform does not suit me, it's a pity. I will consider other services. Good luck and prosperity to you!
